### PR TITLE
website: Terraform v0.15 upgrade guide first draft

### PR DESCRIPTION
**Pull request by apparentlymart**
As usual, we'll continue to iterate on this based on feedback and questions during the beta period, but this is a first draft intended to help those who are trying out the first beta.

[GitHub-rendered Version](https://github.com/hashicorp/terraform/blob/doc-v0.15-upgrade/website/upgrade-guides/0-15.html.markdown) (which will become invalid once this is merged and thus I've deleted the PR branch.)

### CODE REVIEW

**Comment by apparentlymart** [comment]
The link check failure here is because we don't yet have a v0.15 branch to hold the v0.15 CHANGELOG. That'll get rectified once we make the v0.15 maintenence branch, in advance of publishing v0.15-beta1 from it.

File: website/upgrade-guides/0-15.html.markdown [hunk 1]
```diff
@@ -0,0 +1,378 @@
...
+    }
+  }
+}
+```
+
+Warnings will be emitted now where empty configuration blocks are present but
+no longer required, though they continue to work unchanged in the 0.15 release.
+There are a few cases where existing configurations may return new errors:
+
+* A provider configuration within a module cannot be overridden by supplying a
```
Comment (mildwonkey) [inline website/upgrade-guides/0-15.html.markdown:196]: The subject of these bullets are inconsistent and I found that made this section a bit hard to follow. I think it would flow better if the first sentence of each bullet was more consistent. I like the last bullet, which is in the form "<thing> is now an error". since it matches your introductory sentence.

How does that sound to you?

File: website/upgrade-guides/0-15.html.markdown [hunk 2]
```diff
@@ -0,0 +1,378 @@
...
+
+Previous versions of Terraform accommodated this by using an API translation
+layer which could convert a subset of typical virtual terminal sequences into
+corresponding Windows Console API function calls, but as a result this has
+prevented Terraform from using more complex terminal features such as progress
+indicators that update in place, menu prompts, etc.
+
+Over the course of several updates to Windows 10, Microsoft has introduced
+virtual terminal support similar to other platforms and
+[now recommends the virtual terminal approach for application developers].
```
Comment (mildwonkey) [inline website/upgrade-guides/0-15.html.markdown:307]: is this missing its link?

**Review by mildwonkey — APPROVED** [review]
This looks great! There's a missing link, but I'll approve now and let you fix it, and I left one editorial comment which you can respond to or ignore as you see fit.

File: website/upgrade-guides/0-15.html.markdown (hunk 1 above)
Comment (apparentlymart) [reply website/upgrade-guides/0-15.html.markdown]: It looks like @jbardin updated this part, since it was in the commit he'd added. I'm going to run with that updated version for the moment, cause we'll be doing at least a couple more editorial passes of those whole thing before v0.15.0 final. Thanks!
